underscore.js

```diff
@@ -1115,6 +1115,15 @@
     if (a === b) return a !== 0 || 1 / a === 1 / b;
     // A strict comparison is necessary because `null == undefined`.
     if (a == null || b == null) return a === b;
+    // `NaN`s are equivalent, but non-reflexive.
+    if (a !== a) return b !== b;
+    // Exhaust primitive checks
+    var type = typeof a;
+    if (type !== 'function' && type !== 'object' && typeof b !== 'object') return false;
```

Comment:
In ld I simplified the check to:
```js
if (value == null || other == null || (!isObject(value) && !isObject(other))) {
  return value !== value && other !== other;
}
```

which squishes the `NaN` and nullish checks into 1 condition.